src/Meilisearch/MultiSearchResult.cs

```diff
@@ -0,0 +1,18 @@
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Meilisearch
+{
+    /// <summary>
+    /// Search result used in multi-index search
+    /// </summary>
+    public class MultiSearchResult
+    {
+        /// <summary>
+        /// The search results
+        /// </summary>
+        [JsonPropertyName("results")]
+        public List<ISearchable<JsonDocument>> Results { get; set; }
```

Comment:
I'm pretty sure it's not doable, but why can't we do something like:
```
public class MultiSearchResult<T>
...
public List<ISearchable<T>> Results { get; set; }
```
For after the fact, just do something like:
```
            foreach (var item in result.Results)
            {
                if (item is PaginatedSearchResult<Movie> paginated) {
                    /// ... Do something with your movie ...
                }
                if (item is SearchResult<Movie> normal) {
                    /// ... Do something with your movie ...
                }
            }
```

Reply:
we can of course go down the rabbit hole of `OneOf<T1,T2,...>` but that would add more unnecessary complexity